engine/class_modules/priest/sc_priest_holy.cpp

```diff
@@ -164,52 +407,47 @@ action_t* priest_t::create_action_holy( util::string_view name, util::string_vie
   {
     return new holy_fire_t( *this, options_str );
   }
-
   if ( name == "apotheosis" )
   {
     return new apotheosis_t( *this, options_str );
   }
-
   if ( name == "holy_word_chastise" )
   {
     return new holy_word_chastise_t( *this, options_str );
   }
-
   if ( name == "holy_word_serenity" )
   {
     return new holy_word_serenity_t( *this, options_str );
   }
-
-  if ( name == "renew" )
+  if ( name == "holy_word_sanctify" )
   {
-    return new renew_t( *this, options_str );
+    return new holy_word_sanctify_t( *this, options_str );
   }
-
-  if ( name == "heal" )
+  if ( name == "empyreal_blaze" )
+  {
+    return new empyreal_blaze_t( *this, options_str );
+  }
+  if ( name == "divine_word" )
+  {
+    return new divine_word_t( *this, options_str );
+  }
+  if ( name == "searing_light" )
   {
-    return new holy_heal_t( *this, options_str );
+    return new searing_light_t( *this );
   }
 
   return nullptr;
 }
 
-expr_t* priest_t::create_expression_holy( action_t*, util::string_view /*name_str*/ )
+void priest_t::init_background_actions_holy()
 {
-  return nullptr;
+  background_actions.holy_fire     = new actions::spells::holy_fire_t( *this, "" );
```

Comment:
Same thing for holy fire.
```suggestion
  if ( talents.holy.divine_word.enabled() )
  {
  background_actions.holy_fire     = new actions::spells::holy_fire_t( *this, "" );
  background_actions.holy_fire -> background = true;
  }
```

In smite_t you can then do
```
smite_t(...) : child_holy_fire(priest().background_actions.holy_fire) ...
```
instead of first initializing it to nullptr and then setting it in the ctor body.

Reply:
Awesome insight, revised!